### PR TITLE
Converted to V3, added VX1000 layer update action

### PR DESCRIPTION
**Pull request by cyberblaststudios**
- Added VX1000 Layer Update action, which allows for layer configuration and PIP functionality on the VX1000 device.
- Converted the module to Companion V3.0

### CODE REVIEW

File: package.json
Comment (dnmeid) [inline package.json:1]: Many of the properties can be stripped out of package.json. They should only be defined in one place (single point of truth).

File: package.json [hunk 1]
```diff
@@ -35,5 +35,10 @@
 	"bugs": {
 		"url": "https://github.com/bitfocus/companion-module-novastar-controller/issues"
 	},
-	"homepage": "https://github.com/bitfocus/companion-module-novastar-controller#readme"
+	"homepage": "https://github.com/bitfocus/companion-module-novastar-controller#readme",
+	"dependencies": {
+		"@companion-module/base": "~1.0",
+		"@companion-module/tools": "^1.2.0",
+		"yarn": "^1.22.19"
```
Comment (dnmeid) [inline package.json:42]: Why is yarn a dependency? I don't see where the module uses yarn.

File: config.js
Comment (dnmeid) [inline config.js:1]: suggest to rename this file to choices.js

File: actions.js [hunk 2]
```diff
@@ -1,157 +1,391 @@
...
+
+	// Priority
+	bufferArray.push(layerPriorityBuffer);
+
+	// Source (connector code)
+	bufferArray.push(connectorCodeBuffer);
+
+	// initial X position
+	const initialXBuffer = Buffer.allocUnsafe(4);
+	initialXBuffer.writeInt32LE(initialX)
```
Comment (dnmeid) [inline actions.js:54]: write will fail if initialX is no integer in valid range. You should round it to integer here and clamp the range in the action option.
Same for the other buffer assignments.

File: actions.js [hunk 3]
```diff
@@ -1,157 +1,391 @@
...
+				instance.socket.send(element.cmd);
+			}
 		};
 
 		actions['take'] = {
-			label: 'Take Preview to Program'
+			name: 'Take Preview to Program',
+			options: [],
+			callback: async (event) => {
+				var cmd = Buffer.from([0x55,0xAA,0x00,0x00,0xFE,0x00,0x00,0x00,0x00,0x00,0x01,0x00,0x2D,0x00,0x00,0x13,0x01,0x00,0x00,0x95,0x56]);
```
Comment (dnmeid) [inline actions.js:383]: use let or const instead of var

**Review by dnmeid — CHANGES_REQUESTED** [review]
I've put some comments to the files and code. Please also check formatting, there is inconsistent use of semicolons. Add newline to end of files as well.

**Comment by cyberblaststudios** [comment]
@dnmeid made the requested changes, but I will need to test it on the VX1000 first before merging. I will let you know when I have tested. I'll reply to the requested changes where I have questions.

File: package.json
Comment (cyberblaststudios) [reply package.json:1]: @dnmeid I committed a change where I have removed most of the duplication. Not sure what all is required to be in the package.json file. I did keep the version, api_version, dependencies, etc. in the package.json file. Not sure if these can just be in the manifest. But I did clean it up quite a bit.

**Comment by cyberblaststudios** [comment]
> I've put some comments to the files and code. Please also check formatting, there is inconsistent use of semicolons. Add newline to end of files as well.

I also added prettier formatting to this module and formatted every file but choices.js accordingly.

**Comment by cyberblaststudios** [comment]
@dnmeid  Was at church this morning and tested on the VX1000, and we should be good to go except for my one question about the package.json.

File: package.json
Comment (dnmeid) [reply package.json:1]: only the properties needed by yarn are mandatory in the package.json, so api_version could also be omitted, but it is not a big problem if this is still there atm and is removed in a later update

File: package.json [hunk 4]
```diff
@@ -1,39 +1,15 @@
 {
-	"name": "novastar-controller",
-	"legacy": [
-		"novastar-vx4s"
-	],
-	"version": "1.0.8",
+	"version": "2.0.0",
```
Comment (dnmeid) [inline package.json:2]: name is mandatory and can't be omitted
```suggestion
       "name": "novastar-controller",
	"version": "2.0.0",
```

**Review by dnmeid — CHANGES_REQUESTED** [review]

**Pull request merged**